### ISSUE TITLE
Issues with draft Windows installer

### ISSUE THREAD

**Issue by MarkCallow** [KhronosGroup/KTX-Software#244]
I've had the following issues with the draft Windows installer:

- [ ]  1. The installer name has _Windows_ in it. It needs to indicate whether it is for 32- or 64-bit Windows.
- [ ]  2. Warning from Win 10 requiring bypass to install software from an unidentified developer. (I.e. we need to sign it.)
- [x] 3. Norton antivirus accuses the installer of suspicious activity (SONAR.Heuristic.170) before it even displays its first window. A similar report said the file triggering it was a .ini fle in an installer. That was a false positive as I suspect this is. I've reported it to, er, Broadcom, who seems to have bought Symantec who years ago bought Norton.
- [ ] 4. Attempt to add the install directory to %PATH% fails saying the PATH is too long. It isn't. The maximum supported length is 2048 and my %PATH% is around 1968 characters so there is room for `c:\Program Files\KTX-Software\bin`.
- [ ] 5. Application version numbers have "-dirty" appended.
- [ ] 6. The license display does not understand rtf. The text is large and links aren't displayed correctly.

**Comment by MarkCallow** [comment]
It is necessary to also sign the uninstaller. Since this is normally created during the install, special steps are needed, outlined in https://nsis.sourceforge.io/Signing_an_Uninstaller.

**Comment by MarkCallow** [comment]
No. 5 is because the OpenGL ES emulator is being loaded into a directory in the git workarea. Either put the following outside the workarea else which is preferable, or add a .gitignore rule.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	libEGL.dll
	libEGL.lib
	libGLES_CM.dll
	libGLES_CM.lib
	libGLESv2.dll
	libGLESv2.lib

**Comment by MarkCallow** [comment]
Unbelievably weird. Cmd prompt continues to parse lines starting with REM and if it sees &, it executes what follows. I had to put another REM after the & to stop the 32-bit download.

**Comment by MarkCallow** [comment]
Norton, er ... Broadcom, has confirmed the false positive and say the detection will be removed from their products in the next virus definitions update. The whitelisting may take 24 hours to take effect.

**Comment by MarkCallow** [comment]
I can confirm that Broadcom has fixed item 3. Norton users, and probably Symantec users too, will have to use LiveUpdate to get the latest definitions with the fix.

**Comment by zeux** [comment]
One more small issue is that after installation, KTX-Software folder in Program Files has an executable file with an empty name (probably the uninstaller?)

![image](https://user-images.githubusercontent.com/1106629/86489522-7439d280-bd19-11ea-8298-0764ab274b0d.png)

**Comment by MarkCallow** [comment]
Re adding directory to %PATH%. I was able to manual add it, i.e. via the Windows control panel so there was definitely space. This needs further investigation.

**Comment by MarkCallow** [comment]
The latest installer has fixed a couple of the issues and I've checked them off.

Issue #3 is still a big problem. Still coming despite multiple reports to Norton. Norton says that the suspicious behavior includes starting an event watcher process and starting an event process. Why would NSIS be starting an event watcher? It does sound suspicious. Any ideas @atteneder? Is this normal behavior for an installer?

**Comment by atteneder** [comment]
> The latest installer has fixed a couple of the issues and I've checked them off.

Great!

> Issue #3 is still a big problem. Still coming despite multiple reports to Norton. Norton says that the suspicious behavior includes starting an event watcher process and starting an event process. Why would NSIS be starting an event watcher? It does sound suspicious. Any ideas @atteneder? Is this normal behavior for an installer?

No, not that I know of. But I don't use anti virus atm, so no idea.

**Comment by MarkCallow** [comment]
I suspect that the installer is using Windows' `setx` command to add the KTX `bin` directory to %PATH%. This has the unfortunate behavior of truncating the target value (i.e. PATH) when it is > 1025 bytes long. The warning most likely comes because NSIS is checking if the length will exceed this. Apart from the `setx` behavior this is all guess work.
